### PR TITLE
#181 Easier dependency management

### PR DESCRIPTION
**Pull request by JoschD**
* Changed dependency layout
* added 'all'
* some pytest-args examples

Controversial:
* added pillow-requirement as old version kept crashing with matplotlib (https://github.com/matplotlib/matplotlib/issues/14263 )

### CODE REVIEW

**Comment by JoschD** [comment]
I also removed the main-dependencies from the `tests_require` as they are installed anyway, so it was redundant.

File: setup.py [hunk 1]
```diff
@@ -41,6 +49,7 @@ def run_tests(self):
 # Dependencies for the package itself
 DEPENDENCIES = [
     "matplotlib>=3.1.0",
+    "pillow>=6.0.0",  # not our dependency but older versions crash with mpl
```
Comment (fsoubelet) [inline setup.py:52]: The actually package is `Pillow`, though I don't know if the lowercase P could be an issue. Can't hurt to make the change.

File: setup.py [hunk 2]
```diff
@@ -53,19 +62,26 @@ def run_tests(self):
...
+        "attrs>=19.2.0",
+    ],
+    "doc": [
+        "sphinx",
+        "travis-sphinx",
+        "sphinx_rtd_theme"
+    ],
+}
+EXTRA_DEPENDENCIES.update(
+    {'all': [elem for list_ in EXTRA_DEPENDENCIES.values() for elem in list_]}
```
Comment (fsoubelet) [inline setup.py:83]: Some verbosity can't hurt. Maybe `[requirement for extra_reqs in EXTRA_DEPENDENCIES.values() for requirement in extra_reqs]`?

**Comment by JoschD** [comment]
From previous travis runs it looks like python 3.8 does not like mpl 3.1.0 in this combo. Shall we specify mpl == 3.2.* ?

File: setup.py (hunk 1 above)
Comment (fsoubelet) [reply setup.py:52]: @JoschD I just saw that matplotlib has a `Pillow>=6.2.0"` requirement, is this one needed then?

**Comment by JoschD** [comment]
>to include the README changes

They are already merged

File: setup.py (hunk 2 above)
Comment (JoschD) [reply setup.py:83]: I don't think it makes it any more readable and just makes the line much longer.

**Comment by fsoubelet** [comment]
> They are already merged

My bad.

**Comment by lmalina** [comment]
> From previous travis runs it looks like python 3.8 does not like mpl 3.1.0 in this combo. Shall we specify mpl == 3.2.* ?
 I guess, we should specify matplotlib >=3.2.0, this is also a good PR to check if the other minimal versions are still valid.
I am certain about pandas. At some point I had troubles with numpy, so maybe numpy >=1.16.5, which was a solution, etc.

File: setup.py (hunk 1 above)
Comment (JoschD) [reply setup.py]: Weird. I have an old mpl and old Pillow (5.1.0) installed. 
Only the mpl requirement in the Dependencies.
Then I run `python setup.py install test`.
-> Installs mpl 3.2.1
crashes `ImportError: cannot import name '_imaging' from 'PIL' (/usr/lib/python3/dist-packages/PIL/__init__.py)`

With the explicit pillow dependency it updates pillow and runs fine.
Can you verify?

**Comment by JoschD** [comment]
> I guess, we should specify matplotlib >=3.2.0, this is also a good PR to check if the other minimal versions are still valid.
@lmalina 
I think we should be careful with >=   ...  I'd stay on the current minor version, as higher versions might break things? Or go with >= and < and then change the limits after being tested

**Comment by lmalina** [comment]
> > I guess, we should specify matplotlib >=3.2.0, this is also a good PR to check if the other minimal versions are still valid.
> 
> @lmalina
> I think we should be careful with >= ... I'd stay on the current minor version, as higher versions might break things? Or go with >= and < and then change the limits after being tested

I would not be too restrictive towards future versions. Of course, it is to be tested, but that is generally the case, for example, in travis. We find out quite soon if something breaks with newer version. In such cases, we can put upper limit, unless we fix it right away.

File: setup.py (hunk 1 above)
Comment (fsoubelet) [reply setup.py]: Indeed I can confirm, quite strange. Let's leave the Pillow requirement here.

**Comment by JoschD** [comment]
> > > I guess, we should specify matplotlib >=3.2.0, this is also a good PR to check if the other minimal versions are still valid.
> > 
> > 
> > @lmalina
> > I think we should be careful with >= ... I'd stay on the current minor version, as higher versions might break things? Or go with >= and < and then change the limits after being tested
> 
> I would not be too restrictive towards future versions. Of course, it is to be tested, but that is generally the case, for example, in travis. We find out quite soon if something breaks with newer version. In such cases, we can put upper limit, unless we fix it right away.

I updated the lower versions. These are the ones I found (starting from what we had before) that will work on python 3.6-3.8 if you go with a vanilla python dist. and use exactly theses versions. Lower versions crashed for one reason or another.

**Pull request merged**